code/dataobjects/Comment.php

```diff
@@ -267,54 +267,80 @@ public function canCreate($member = null) {
 	 * @return Boolean
 	 */
 	public function canView($member = null) {
-		if(!$member) $member = Member::currentUser();
+		$member = $this->getMember($member);
 
-		// Standard mechanism for accepting permission changes from decorators
-		$extended = $this->extendedCan('canView', $member);
-		if($extended !== null) return $extended;
+		if(!$member) {
+			return false;
+		}
 
-		// Allow admin
-		if(Permission::checkMember($member, 'CMS_ACCESS_CommentAdmin')) return true;
+		if($extended = $this->extendedCan('canView', $member)) {
```

Comment:
We need to consider that `$extended` could be `false` as well. The existing check is a tri-state null = no override, false = can't view, true = can edit.